layouts/partials/scripts.html

```diff
@@ -11,18 +11,28 @@
 <script src="{{ $jsBootstrap.RelPermalink }}" type="text/javascript" integrity="{{ $jsBootstrap.Data.Integrity }}"></script>
 
 
-{{ if ( not ( in .Site.Params.buildtype "package" ) ) }}
+
 <!-- START COVEO -->
 <script class="coveo-script" src="https://static.cloud.coveo.com/searchui/v2.10104/0/js/CoveoJsSearch.Lazy.min.js" integrity="sha512-HxdDBIp5snbqtu1TPkBnXLKEvN9IPz3PeZW8zT9KfsflV9Ck822XDroDlpVbfCfXeiu8C0RDVn0fdu7aGy/H1g==" crossorigin="anonymous"></script>
 
 <script src="https://static.cloud.coveo.com/searchui/v2.10104/0/js/templates/templates.js" integrity="sha512-CR0Yk/LIwgh1MsKqjecDp/r6F9ipKc6gA+4+E1pplT3N3r1pk+la/HaqyDiKtjOFwrwIIbIYBFrUJgPql93QHw==" crossorigin="anonymous"></script>
 
-{{ $coveo := resources.Get "js/coveo.js" }}
-{{ $secureCoveo := $coveo | minify | fingerprint "sha512" }}
+<!-- build a different coveo.js (with different key) based on build env -->
+{{- $options := dict "targetPath" "js/coveo.js" -}}
+{{ $.Scratch.Set "coveoSc" "" }}
+{{ if eq hugo.Environment "production" }}
+  {{- $coveo := resources.Get "js/coveo-prod.js" | js.Build $options -}}  
+  {{ $.Scratch.Set "coveoSc" $coveo}}
```

Comment:
There are 3 coveo environments -- test, staging, and production. Please add the staging API key here as an option as well.